Atomizer/src/main/java/senghout/github/com/atomizer/AtomizerController.java

```diff
@@ -35,13 +40,38 @@ public String getUrl(@PathVariable("TinyUrl")String tinyUrl) {
 
     @PostMapping(value = "/add", consumes = {"application/json"})
     public String addUrl(@RequestBody AddUrlInput data) {
-        if (zoo == null || zoo.low == zoo.high) {
-            zoo = heimdall.getNextRange();
+        if (zoo == null || this.index == order.length) {
+            setupNextRange();
         }
-        final String tinyUrl = atomizerUtils.encodeNumber(zoo.low++);
+        long nextValue = generateSeedUrlValue();
+        final String tinyUrl = atomizerUtils.encodeNumber(nextValue);
         final TinyUrl tiny = new TinyUrl(tinyUrl, data.fullUrl);
         repo.save(tiny);
 
         return tinyUrl;
     }
+
+    private void setupNextRange() {
+        this.zoo = heimdall.getNextRange();
+        this.order = new int[(int)(zoo.high - zoo.low)];
```

Comment:
1 . `zoo.high - zoo.low` could be larger than `Integer.MAX_VALUE`
2. we use `long[SOME_LENGTH]` instead of `int[]` than have `SOME_LENGTH = 1,000,000` or sth

Reply:
Unfortunately this will be a limitation, long[] arr = new long[SOME_LENGTH], SOME_LENGTH still has to be an int

Reply:
hence the `SOME_LENGTH = 1,000,000`

Reply:
Ah I see your point, I think I'll keep it the same though. It's more flexible this way. If we ever decide to change the range (never) we'll have to remember to change the limit here too.

Reply:
lol alrite i'll log it as known issue for now